src/types/spot.ts

```diff
@@ -404,29 +408,40 @@ export interface ExchangeInfoParams {
   symbols?: string[];
 }
 
-export interface NewSpotOrderParams {
+export interface NewSpotOrderParams<
+  T extends OrderType,
```

Comment:
Any idea how we can make TypeScript automatically infer the order type from usage? For example, this is complaining because the first parameter of the generic type is missing:
<img width="1177" alt="Screenshot 2024-05-22 at 12 18 43" src="https://github.com/tiagosiebler/binance/assets/14090872/ac61c308-0e72-4942-9308-9ade80fab099">

However, it was never required before and would be a slight breaking change to require it now. It also seems redundant to force this:
```
      const order: NewSpotOrderParams<'MARKET'> = {
        symbol,
        side: 'BUY',
        type: 'MARKET',
        quantity: 0.1,
      };
```

When the value in the object already says exactly what the order type is. Any idea how to make TypeScript automatically infer TOrderType in a scenario like this? It's common to generate a request param (like an order object) before actually calling the function.

Reply:
This is much better, thanks! This way there's no impact for anyone upgrading, just improvements/more flexibility. 

Can you check if any other interfaces you've updated need to be updated as well? Or is this the only one?

Reply:
There were four more, and I fixed them in the last commit.